src/next_step_runner/main/main.go

```diff
@@ -4,10 +4,12 @@ import (
 	"github.com/openshift/assisted-installer-agent/src/commands"
 	"github.com/openshift/assisted-installer-agent/src/config"
 	"github.com/openshift/assisted-installer-agent/src/util"
+	log "github.com/sirupsen/logrus"
 )
 
 func main() {
 	config.ProcessArgs()
 	util.SetLogging("agent_next_step_runner", config.GlobalAgentConfig.TextLogging, config.GlobalAgentConfig.JournalLogging)
 	commands.ProcessSteps()
+	log.Info("Stopping to process steps and exiting")
```

Comment:
agent will not be just restarted? or this is what u want?

Reply:
It might be restarted by the process that runs it, if needed. I want next_step_runner to exit when it receives a flag in an server response to do so.

Reply:
Consider changing the log to:
```suggestion
	log.Info("next step runner exiting")
```